flink-core/src/main/java/org/apache/flink/configuration/SecurityOptions.java

```diff
@@ -490,4 +490,25 @@
                                     + "channel. If the `close_notify` was not flushed in the given timeout the channel will be closed "
                                     + "forcibly. (-1 = use system default)")
                     .withDeprecatedKeys("security.ssl.close-notify-flush-timeout");
+
+    // ------------------------ security manager --------------------------------
+
+    /** Flag to check the attempt of main or user-defined function to terminate JVM. */
+    @Documentation.Section(Documentation.Sections.EXPERT_DEBUGGING_AND_TUNING)
+    public static final ConfigOption<String> CHECK_SYSTEM_EXIT =
+            key("security.check-system-exit")
```

Comment:
As commented already, I suggest to move this to the `cluster.` config namespace. I don't see this as a security setting, rather controlling the behavior of the process.